### DISCUSSION TITLE
Is there a "pre-create-command"?

### DISCUSSION THREAD

**Discussion by enboig** [rustic-rs/rustic#1200] [Q&A]
I know it can be used in a script, but seeing `post-create-command` I hoped I could add all inside profile.toml file.

I just want to dump database before backup.

**Comment by enboig** [comment]
I checked the code and "post-create-command" is only used in 'impl WriteBackend for LocalBackend {'; does it mean it isn't run for other repositories (like openDAL, Rclone, ...)?

I know the example uses it to create PAR files; but user cases aren't limited to it.

Could both "pre" and "post" be run for all backends? and let user configure when they have sense?

**Comment by aawsome** [comment]
I think we must distinguish 2 situations here:
- command(s) which are called whenever a file is written to the backend. This is currently handled by the `post-create` command but only implemented for the local backend. So, yes, we can implement this for all backends.
- command(s) which are called before or after a rustic command is performed, e.g. pre-backup or post-backup. This is what we are currently working on, see https://github.com/rustic-rs/rustic/pull/1218
